### PR TITLE
Add duration constants

### PR DESCRIPTION
**Pull request by ghost**
Add constants `SECOND`, `MILLISECOND`, `MICROSECOND`, and `NANOSECOND` to `core::time`.

This will make working with durations more ergonomic. Compare:

```rust
// Convenient, but deprecated function.
thread::sleep_ms(2000);

// The current canonical way to sleep for two seconds.
thread::sleep(Duration::from_secs(2));

// Sleeping using one of the new constants.
thread::sleep(2 * SECOND);
```

### CODE REVIEW

**Comment by stjepang** [comment]
Previous attempts to do something similar, but got stuck on bikeshedding: https://github.com/rust-lang/rust/pull/52556 https://github.com/rust-lang/rust/pull/51610

Since we already have `std::time::UNIX_EPOCH`, I hope adding a few more useful constants won't be too controversial.

**Review by sugar700 — APPROVED** [review]

**Comment by retep998** [comment]
Shouldn't we create a tracking issue for this first?

**Comment by frewsxcv** [comment]
https://github.com/rust-lang/rust/issues/57391

**Comment by xfix** [comment]
I think the branch should be updated with tracking issue number.

File: src/libcore/time.rs [hunk 1]
```diff
@@ -23,6 +23,22 @@ const MILLIS_PER_SEC: u64 = 1_000;
 const MICROS_PER_SEC: u64 = 1_000_000;
 const MAX_NANOS_F64: f64 = ((u64::MAX as u128 + 1)*(NANOS_PER_SEC as u128)) as f64;
 
+/// The duration of one second.
+#[unstable(feature = "duration_constants", issue = "0")]
```
Comment (xfix) [inline src/libcore/time.rs]: The issue number should be updated for the tracking issue.

**Review by sugar700 — CHANGES_REQUESTED** [review]

File: src/libcore/time.rs (hunk 1 above)
Comment (joshtriplett) [reply src/libcore/time.rs]: Ah, good call. @stjepang, can you please update those to reference the tracking issue?

Comment (stjepang) [reply src/libcore/time.rs]: Done.

**Review by sugar700 — APPROVED** [review]

**Pull request merged**